README.md

```diff
@@ -1,20 +1,262 @@
-Cliquetis
-=========
+# Cliquetis
 
 [![Build Status](https://travis-ci.org/mozilla-services/cliquetis.svg?branch=master)](https://travis-ci.org/mozilla-services/cliquetis)
 
 A JavaScript client for [Cliquet](https://github.com/mozilla-services/cliquet/).
 
-Installation
-------------
+This is work in progress, there's no code just yet.
+
+## Installation
 
     $ npm install
 
-Build
------
+## Build
 
     $ npm run dist
 
+## Usage
+
+**Caution: everything in this section is still pure fiction.**
+
+* Every operation is performed locally;
+* Synchronization with server shall be ran explicitly.
+
+### The `Cliquetis` constructor
+
+```js
+var db = new Cliquetis(options);
+```
+
+`options` is an object defining the following option values:
+
+- `remote`: The remote Cliquet server endpoint root URL (eg. `"https://server/v1"`);
+- `headers`: The default headers to pass for every HTTP request performed to the Cliquet server (eg. `{"Authorization": "Basic bWF0Og=="}`);
+- `mode`: The conflict default resolution strategy (`Cliquet.SAFE` (*server wins*), `Cliquet.FORCE` (*client wins*)).
+
+### Collection
+
+Selecting a collection is done by calling the `collection()` method, passing it the resource name:
+
+```js
+var articles = db.collection("articles");
+```
+
+This collection object has the following attributes:
+
+* **lastModified**: last synchronization timestamp, ``null`` if never sync'ed.
+
+> Synchronization timestamps are persisted in the device local storage.
+
+All operations are asynchronous and rely on [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+
+### Creating a record
+
+```js
+articles.save({title: "foo"}).then(console.log.bind(console));
+```
+
+Result is:
+
+```js
+// result
+{
+  data: {
+    id: "2dcd0e65-468c-4655-8015-30c8b3a1c8f8",
+    title: "foo",
+  }
+}
+```
+
+> Records identifiers are generated locally using UUID4.
+
+### Retrieving a single record
+
+```js
+articles.get("2dcd0e65-468c-4655-8015-30c8b3a1c8f8")
+  .then(console.log.bind(console));
+```
+
+Result:
+
+```js
+{
+  data: [
+    {
+      id: "2dcd0e65-468c-4655-8015-30c8b3a1c8f8",
+      title: "bar"
+    }
+  ]
+}
+```
+
+### Updating a record
+
+```js
+var existing = {
+  id: "2dcd0e65-468c-4655-8015-30c8b3a1c8f8",
+  title: "bar"
+};
+
+var updated = Object.assign(existing, {
+  title: "baz"
+});
+
+articles.save(updated).then(console.log.bind(console));
+```
+
+Result is:
+
+```js
+{
+  data: {
+    id: "2dcd0e65-468c-4655-8015-30c8b3a1c8f8",
+    title: "baz",
+    last_modified: 1432222889337
+  }
+}
+```
+
+### Deleting records
+
+#### Single unique record passing its `id`:
+
+```js
+articles.delete("2dcd0e65-468c-4655-8015-30c8b3a1c8f8")
```

Comment:
It seems counter intuitive: for all other operations, we're passing an object and for this one we're passing an id